server/data_cxg/cxg_adaptor.py

```diff
@@ -250,6 +250,9 @@ def get_embedding_names(self):
         with ServerTiming.time(f"layout.lsuri"):
             pemb = self.get_path("emb")
             embeddings = [os.path.basename(p) for (p, t) in self.lsuri(pemb) if t == "array"]
+        if len(embeddings) == 0:
+            # something went wrong. perhaps the dataset was removed.  re-validate
+            self.pre_load_validation(self.data_locator)
```

Comment:
is there ever a case where pre_load_validation will not throw?  At which point the embedding are still potentially wrong, OR the dataset actually has no embeddings (which is rare, but in theory could occur)

If we know this is a hard error, we should just raise it here.  If it is spurious, and embeddings will exist, then they need to reload/retry.   If embeddings are optional, then the real issue is that the caller needs to handle this check (ie, /schema endpoint needs to do its own validation that the object exists).

Today, embeddings are required in CXG, as they are an essential part of the app.  So I recommend you just throw an error here....

Reply:
Sounds good.  fixed.